src/SiteProcess.php

```diff
@@ -92,21 +94,26 @@ protected function processArgs()
         return $transport->wrap($processedArgs);
     }
 
-    /**
-     * TODO: Could we perhaps support variable transport mechanisms?
-     */
-    protected static function getTransport(AliasRecord $siteAlias)
+    public function setTransportManager($transportManager)
```

Comment:
Why aren't you adding type hints for parameters (in doc blocks and parameters) and return types?. This line could be changed to ensure $transportManager is/extends TransportManager